default.nix

```diff
@@ -70,10 +70,12 @@ let
 
         configureFlags = with lib.lists; oldAttrs.configureFlags ++ [
           "--enable-mpi"
-          "MPICC=${self_.mpi}/bin/mpicc"
-          "MPIFC=${self_.mpi}/bin/mpif90"
-          "MPIF90=${self_.mpi}/bin/mpif90"
+          "MPICC=${self.mpi}/bin/mpicc"
+          "MPIFC=${self.mpi}/bin/mpif90"
+          "MPIF90=${self.mpi}/bin/mpif90"
         ];
+
+        passthru.mpi = { mpi = self.mpi; };
```

Comment:
For a library it is not really needed. Putting `mpi` in `propagatedBuildInputs` is probably more useful in this case.